test/src/main/java/org/apache/accumulo/test/fate/MultipleStoresIT.java

```diff
@@ -0,0 +1,553 @@
+      assertTrue(allIds.add(store1.create()));
+    }
+    assertEquals(numFateIds, allIds.size());
+
+    // Reserve half the fate ids using store1 and rest using store2, after reserving a fate id in
+    // one, should not be able to reserve the same in the other. Should also not matter that all the
+    // ids were created using store1
+    int count = 0;
+    for (FateId fateId : allIds) {
+      if (count % 2 == 0) {
+        reservations.add(store1.reserve(fateId));
+        assertTrue(store2.tryReserve(fateId).isEmpty());
+      } else {
+        reservations.add(store2.reserve(fateId));
+        assertTrue(store1.tryReserve(fateId).isEmpty());
+      }
+      count++;
+    }
+    // Try to reserve a non-existent fate id
+    assertTrue(store1.tryReserve(fakeFateId).isEmpty());
+    assertTrue(store2.tryReserve(fakeFateId).isEmpty());
+    // Both stores should return the same reserved transactions
+    activeReservations = store1.getActiveReservations();
+    assertEquals(allIds, activeReservations.keySet());
+    activeReservations = store2.getActiveReservations();
+    assertEquals(allIds, activeReservations.keySet());
+
+    // Test setting/getting the TStatus and unreserving the transactions
+    for (int i = 0; i < allIds.size(); i++) {
+      var reservation = reservations.get(i);
+      assertEquals(ReadOnlyFateStore.TStatus.NEW, reservation.getStatus());
+      reservation.setStatus(ReadOnlyFateStore.TStatus.SUBMITTED);
+      assertEquals(ReadOnlyFateStore.TStatus.SUBMITTED, reservation.getStatus());
+      reservation.delete();
+      reservation.unreserve(Duration.ofMillis(0));
+      // Attempt to set a status on a tx that has been unreserved (should throw exception)
+      assertThrows(IllegalStateException.class,
+          () -> reservation.setStatus(ReadOnlyFateStore.TStatus.NEW));
+    }
+    assertTrue(store1.getActiveReservations().isEmpty());
+    assertTrue(store2.getActiveReservations().isEmpty());
+  }
+
+  @Test
+  public void testReserveNonExistentTxn() throws Exception {
+    testReserveNonExistentTxn(FateInstanceType.META);
+    testReserveNonExistentTxn(FateInstanceType.USER);
+  }
+
+  private void testReserveNonExistentTxn(FateInstanceType storeType) throws Exception {
+    // Tests that reserve() doesn't hang indefinitely and instead throws an error
+    // on reserve() a non-existent transaction.
+    final FateStore<SleepingTestEnv> store;
+    final boolean isUserStore = storeType.equals(FateInstanceType.USER);
+    final String tableName = getUniqueNames(1)[0];
+    final FateId fakeFateId = FateId.from(storeType, UUID.randomUUID());
+    final ZooUtil.LockID lock = new ZooUtil.LockID("/locks", "L1", 50);
+
+    if (isUserStore) {
+      createFateTable(client, tableName);
+      store = new UserFateStore<>(client, tableName, lock, null);
+    } else {
+      store = new MetaFateStore<>(FATE_DIR, zk, lock, null);
+    }
+
+    assertThrows(IllegalStateException.class, () -> store.reserve(fakeFateId));
```

Comment:
Could verify  error messages contains the fate id.

```suggestion
    var err = assertThrows(IllegalStateException.class, () -> store.reserve(fakeFateId));
    assertTrue(err.getMessage().contains(fakeFateId.canonical()));
```

Reply:
Added